src/JsonSerializationWriter.php

```diff
@@ -452,7 +452,13 @@ public function writeDateIntervalValue(?string $key, ?DateInterval $value): void
             if (!empty($key)) {
                 $this->writePropertyName($key);
             }
-            $res = "P{$value->y}Y{$value->y}M{$value->d}DT{$value->h}H{$value->i}M{$value->s}S";
+            $year = $value->y > 0 ? "%yY" : "";
```

Comment:
```php
$yearFormat = $value->y > 0 ? "%yY" : "";
$monthFormat = $value->m > 0 ? "%mM" : "";
$dayFormat = $value->d > 0 ? '%dD' : "";
$hourFormat = $value->h > 0 ? '%hH' : "";
$minuteFormat = $value->i > 0 ? '%iM' : "";
$secondFormat = $value->s > 0 ? '%sS' : "";
$formatString = "%rP{$yearFormat}{$monthFormat}{$dayFormat}T{$hourFormat}{$minuteFormat}{$secondFormat}";
$result = $value->format($formatString);
```
Note - my preference for descriptive variable names, feel free to dismiss

Reply:
I believe `$result` is descriptive enough in this context.